taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import { useEffect, useState } from "react";
+import AddBoxIcon from '@mui/icons-material/AddBox';
+import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline';
+import * as uuid from "uuid";
 import ISO6391 from 'iso-639-1';
 
 /**
  * Sub-component for rendering translation of an "entry"  
 */
-const ListTranslations = ({ nodeObject, setNodeObject }) => {
-    let renderedTranslations = {}
+const ListTranslations = ({ nodeObject, setNodeObject, originalNodeObject }) => {
 
-    Object.keys(nodeObject).forEach((key) => {
+    const [renderedTranslations, setRenderedTranslations] = useState([]) // Stores state of all tags
+    const [mainLang_renderedTranslations, setMainLang_renderedTranslations] = useState([]) // Stores state of main language's tags
+    const [open, setOpen] = useState(false); // Used for Dialog component
+    const [newLanguageCode, setNewLanguageCode] = useState(''); // Used for storing new LC from Dialog
+    const [isValidLanguageCode, setisValidLanguageCode] = useState(false); // Used for validating a new LC
+    const [btnDisabled, setBtnDisabled] = useState(true) // For enabling or disabling Dialog button
+
+    // Helper functions for Dialog component
+    function handleClose() { setOpen(false); }
+    function handleOpen() { setOpen(true); }
+
+    // Used for addition of a translation language
+    function handleAddTranslation(key) {
+        const newRenderedTranslations = [...renderedTranslations, {'languageCode' : key, 'tags' : []}]
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
         
-        // Get all tags and its corresponding language code
-        // Tagids need to be recomputed, so shouldn't be rendered
-        // Main language isn't considered, since it's rendered separately
- 
-        if (key.startsWith('tags') && 
-            !key.endsWith(nodeObject.main_language) && 
-            !key.includes('ids')) {
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject[key] = [];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Used for deleting a translation language
+    function handleDeleteTranslation(key) {
+        const newRenderedTranslations = renderedTranslations.filter(obj => !(key === obj.languageCode))
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
+        
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
```

Comment:
related with other comments, but here you are defining two related values: `setRenderedTranslations` live in this component, `setNodeObject` lives in the parent. Both seem to hold the same information, why not using the one in the parent?  I'm wondering if this here is actually duplicating the info